src/screens/UserManagement/UserRevamp/NewUserInvitationForm.res

```diff
@@ -0,0 +1,202 @@
+      <div className="flex gap-2 h-fit">
+        {elem.groups
+        ->Array.map(item => {
+          <p
+            className={`py-0.5 px-2 rounded-full bg-gray-200 ${p3RegularTextClass} flex gap-1 items-center`}>
+            <Icon name={item->iconForAccess} size=12 />
+            <span> {(item :> string)->React.string} </span>
+          </p>
+        })
+        ->React.array}
+      </div>
+    </div>
+  }
+}
+module RoleToPermission = {
+  @react.component
+  let make = (~roleInfo: array<UserManagementTypes.userModuleType>, ~roleDict, ~role) => {
+    open LogicUtils
+    let userAcessGroup = roleDict->getDictfromDict(role)->getStrArrayFromDict("groups", [])
+    let (modulesWithAccess, moduleWithoutAccess) = UserUtils.modulesWithUserAccess(
+      roleInfo,
+      userAcessGroup,
+    )
+
+    <div className="flex flex-col gap-8">
+      {modulesWithAccess
+      ->Array.mapWithIndex((elem, index) => {
+        <ModulePermissionRender elem index />
+      })
+      ->React.array}
+      {moduleWithoutAccess
+      ->Array.mapWithIndex((elem, index) => {
+        <ModulePermissionRender elem index customCss="text-grey-200" />
+      })
+      ->React.array}
+    </div>
+  }
+}
+
+module NoteComponent = {
+  @react.component
+  let make = () => {
+    <div className="flex gap-2 items-start justify-start">
+      <Icon name="info-vacent" size=18 className="" customIconColor="!text-gray-400" />
+      // TODO : Change based on selection
+      <span className={`${p3RegularTextClass} text-gray-500`}>
+        {"You can only invite people for 'Hyperswitch US' here. To invite users to another organisation, please switch the organisation."->React.string}
+      </span>
+    </div>
+  }
+}
+
+@react.component
+let make = () => {
+  open APIUtils
+  open LogicUtils
+  open UserUtils
+  open UserManagementHelper
+
+  let getURL = useGetURL()
+  let fetchDetails = useGetMethod()
+  let roleInfo = Recoil.useRecoilValueFromAtom(HyperswitchAtom.moduleListRecoil)
+  let (roleDict, setRoleDict) = React.useState(_ => Dict.make())
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
+  let roleTypeValue =
+    ReactFinalForm.useField(`role_id`).input.value->getStringFromJson("")->getNonEmptyString
+  let (options, setOptions) = React.useState(_ => []->SelectBox.makeOptions)
+  let (dropDownLoaderState, setDropDownLoaderState) = React.useState(_ =>
+    DropdownWithLoading.Success
+  )
+  let {userInfo: {userEntity}} = React.useContext(UserInfoProvider.defaultContext)
+
+  let getMemberAcessBasedOnRole = async _ => {
+    try {
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let url = getURL(
+        ~entityName=USER_MANAGEMENT,
+        ~userRoleTypes=ROLE_ID,
+        ~id=roleTypeValue,
+        ~methodType=Get,
+        ~queryParamerters=Some("groups=true"),
+      )
+      let res = await fetchDetails(url)
+      setRoleDict(prevDict => {
+        prevDict->Dict.set(roleTypeValue->Option.getOr(""), res)
```

Comment:
should we make a copy of this a new dict and then update.

Reply:
Nope as none of the rendering is based on roleDict . its based on roleType which is already in use-effect dependancy